File: C++/060. Permutation Sequence.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    std::string getPermutation(int n, int k) {
        std::vector<int> num(n + 1, 1), offset(n + 1, 1);
        for (int i = 1; i <= n; i++) {
            num[i] = i;
            offset[i] = offset[i - 1] * i;
        }
        std::string res("");
        k = k - 1;
        for (int i = n; i > 0; k = k % offset[i - 1], i--) {
            int index = k / offset[i - 1] + 1;
            res += std::to_string(num[index]);
            num.erase(num.begin() + index);
        }
        return res;
    }
};

File: Go/060. Permutation Sequence.go
package main

import "strconv"

func getPermutation(n int, k int) string {
	num := make([]int, n, n)
	for i := 0; i < n; i++ {
		num[i] = i + 1
	}
	for i := 2; i <= k; i++ {
		var a, b int
		for a = n - 2; a >= 0 && num[a] > num[a + 1]; a-- {}
		for b = a + 1; b < n && num[b] > num[a]; b++ {}
		b--
		num[a], num[b] = num[b], num[a]
		for a, b = a + 1, n - 1; a < b; a, b = a + 1, b - 1 {
			num[a], num[b] = num[b], num[a]
		}
	}
	res := ""
	for i := 0; i < n; i++ {
		res += strconv.Itoa(num[i])
	}
	return res
}